### ISSUE TITLE
Comparison to MoleculeSTM and ProteinDT

### ISSUE THREAD

**Issue by amelie-iska** [QizhiPei/BioT5#6]
Hi! Thanks for the interesting work! I was just wondering if you had and comparisons of BioT5+ to MoleculeSTM or ProteinDT. They are mentioned in the paper but no comparison is given on the text-guided molecule design or text-guided protein design. My intuition is that the contrastive learning CLIP like approach is a great approach to the problem, but it is not clear how it compares to this method and training on multiple modalities sometimes improves performance as well. Do you have a good sense of how the two approaches compare to one another?

**Comment by QizhiPei** [comment]
Thanks for your interest in our work. To my knowledge, MoleculeSTM is primarily evaluated on molecule-text retrieval and editing tasks and not on the text-guided molecule design task. ProteinDT conducts zero-shot text-to-protein generation task with additional decoders, and the evaluation needs well-trained ProteinCLAP model to obtain the retrieval accuracy. So it's possible to compare BioT5/BioT5+ with ProteinDT by further fine-tuning it on the SwissProtCLAP dataset and follow the same evaluation pipeline described in ProteinDT. As ProteinDT is just open source, we haven't done this comparison yet.

From my perspective, CLIP-based model is natural for retrieval task through contrastive learning, but can not directly used for cross-modal generation. The encoder-decoder models (like MolT5, BioT5) or decoder-only model (like MolXPT) are suitable for cross-modal generation but not for retrieval tasks. Features of related molecule/protein-text in such models is not explicitly aligned. Using two BioT5 encoders (one for molecule/protein, one for text) as dual-stream model (like CLIP) may be a choice for retrieval tasks and need further exploration.

**Comment by amelie-iska** [comment]
After reading this, it makes sense now that a comparison to ProteinDT (and MoleculeSTM) was not in the paper. Thank you for the clarification on why.
